assertj-core/src/main/java/org/assertj/core/api/DefaultAssertionErrorCollector.java

```diff
@@ -114,7 +117,7 @@ public List<AssertionError> assertionErrorsCollected() {
    * @since 3.17.0
    */
   public void setAfterAssertionErrorCollected(AfterAssertionErrorCollected afterAssertionErrorCollected) {
-    callback = afterAssertionErrorCollected;
+    callbacks.add(afterAssertionErrorCollected);
```

Comment:
This is misleading to me, I would not expect adding a callback but defining the unique callback, ex:
```java
   softly.setAfterAssertionErrorCollected(err -> stringList.add(err.getMessage()));
    softly.setAfterAssertionErrorCollected(throwableList::add);
```
I expect `throwableList::add` to be the callback and not both of them, like when you do:
```java
person.setName("Jack");
person.setName("Joe");
```
you don't expect `person` to have two names.

For backward compatibility, I would define a new method `addAfterAssertionErrorCollected`.

Reply:
I got it. I'd suggest just rename this method to `addAfterAssertionErrorCollected()`, because according to generally accepted understanding the `set` version should do something like:
```
public void setAfterAssertionErrorCollected(AfterAssertionErrorCollected afterAssertionErrorCollected) {
    callbacks.clear();
    callbacks.add(this);
    callbacks.add(afterAssertionErrorCollected);
}
```
Or even without adding `this`.
But you are right about compatibility, so maby there should be some warning in `javadoc` at least. Something like:

> Same as {@link DefaultAssertionErrorCollector#addAfterAssertionErrorCollected(AfterAssertionErrorCollected)}, but
>  is also deletes all previously set callbacks except the {@code onAssertionErrorCollected()} of {@code this} object.
>Please consider using {@link DefaultAssertionErrorCollector#addAfterAssertionErrorCollected(AfterAssertionErrorCollected)}
>  instead, because another frameworks and integrations can also use this functionality.
> 
>  @param afterAssertionErrorCollected the callback.
> 
>  @since 3.17.0

(this is a draft. I can do better with dictionary and translator :) )
Thus, the previous `javadoc` should be moved to `add` method I suppose.
What do you think?

Reply:
The previous version of the method overwrites even "this.onAssertionErrorCollected()" calllback. So I won't add `callbacks.add(this)` command to `set` method version.